### ISSUE TITLE
Hello, is there way to get all unpacked resources?

### ISSUE THREAD

**Issue by klaxons1** [nenikitov/ashen-decomp#32]

**Comment by nenikitov** [comment]
Hello,

For now, unpacking and parsing are a part of the test suite. I'll clear this up the in the README. 

To get all unpacked resources (will deflate all assets in `output/deflated/` and name them in [this order](https://github.com/nenikitov/ashen-decomp#known-file-declarations))
- Place your ROM in `rom/packfile.dat`
- Run 
  ```sh
  # Specific test that unpacks resources without parsing them
  cargo test --release -- --ignored parse_rom_packfile
  ```
To get some parsed resources (will parse in `output/parsed/`). Hopefully parsing can be better integrated alongside #11
  - Run
    ```sh
    # Run after `parse_rom_packfile`
    cargo test --release -- --ignored parse_rom_asset
    ```

**Comment by UserIsntAvailable** [comment]
Just to clarify (it might not be clear), it is still not possible to get **all** game files on their **parsed** "state" yet. As stated above, #11 (moved to #34) will most likely help with that. [^1]

[^1]: It would probably be worth it to also provide a `binary` wrapper around #34, so that users doesn't need to use library themselves.

**Comment by nenikitov** [comment]
Yes it's still in the testing.

The tests were outputting only 1 file per type (hunter happened to be the model I picked). I changed it though and now the tests should output everything we have parsed so far.

**Comment by nenikitov** [comment]
@RandomTale 
Yes it's still in the testing.

The tests were outputting only 1 file per type (hunter happened to be the model I picked). I changed it though and now the tests should output everything we have parsed so far.

**Comment by RandomTale** [comment]
hmm its still not showing any other models, idk what im doing wrong here. isnt there a way to just get the parsed files?

**Comment by nenikitov** [comment]
@RandomTale what's the file structure after running the tests? It should be
```sh
tree -d output
```
```
output
├── deflated
└── parsed
    ├── color-map
    ├── model
    ├── skybox
    ├── sound
    │   ├── effect
    │   └── song
    ├── string
    └── texture

11 directories
```

**Comment by RandomTale** [comment]
its this

├───deflated
└───parsed
    ├───color-map
    ├───models
    │   └───hunter
    ├───skyboxes
    ├───sounds
    │   ├───effects
    │   └───songs
    ├───strings
    └───textures

**Comment by nenikitov** [comment]
Hmmm.
The PR that made the test suite output all files also changed the naming of directories to be consistently singular, I don't see them.

Are you running the tests from the `main` branch and is it up to date? The last commit should be
```sh
git log -n 1
```
```
commit 12e69767eea2b8ff8e438e0ff4f8841218aede50 (HEAD -> main, origin/main, origin/HEAD)
Author: Mykyta Onipchenko <44075969+nenikitov@users.noreply.github.com>
Date:   Fri May 2 21:52:53 2025 -0400

    test: output of all files (#57)

    ---------

    Co-authored-by: Angel Pineda <57047985+UserIsntAvailable@users.noreply.github.com>
```

**Comment by RandomTale** [comment]
yea its the main one:

[UserIsntAvailable](https://github.com/nenikitov/ashen-decomp/commits?author=UserIsntAvailable)
[test: output of all files (](https://github.com/nenikitov/ashen-decomp/commit/12e69767eea2b8ff8e438e0ff4f8841218aede50)https://github.com/nenikitov/ashen-decomp/pull/57[)](https://github.com/nenikitov/ashen-decomp/commit/12e69767eea2b8ff8e438e0ff4f8841218aede50)
[12e6976](https://github.com/nenikitov/ashen-decomp/commit/12e69767eea2b8ff8e438e0ff4f8841218aede50)
 · 
15 hours ago
---------

Co-authored-by: Angel Pineda <57047985+UserIsntAvailable@users.noreply.github.com>

**Comment by nenikitov** [comment]
Can you try deleting the `output` directory and rerunning the tests?
Also, does running `parse_rom_asset` tests output this (order might differ)?
```
test asset::gamma_table::tests::parse_rom_asset ... ok
test asset::color_map::tests::parse_rom_asset ... ok
test asset::skybox::tests::parse_rom_asset ... ok
test asset::string_table::tests::parse_rom_asset ... ok
test asset::texture::tests::parse_rom_asset ... ok
test asset::model::tests::parse_rom_asset ... ok
test asset::sound::tests::parse_rom_asset ... ok
```

**Comment by RandomTale** [comment]
yep, thats exactly what is says, and yes ive tried running it quite a few times and the output is the same

```
┌──(kali㉿kali)-[~/Desktop/ashen-decomp-main]
└─$ cargo test --release -- --ignored parse_rom_asset
    Finished `release` profile [optimized] target(s) in 0.11s
     Running unittests src/lib.rs (target/release/deps/ashen-bd671703daa93388)

running 7 tests
test asset::color_map::tests::parse_rom_asset ... ok
test asset::gamma_table::tests::parse_rom_asset ... ok
test asset::string_table::tests::parse_rom_asset ... ok
test asset::skybox::tests::parse_rom_asset ... ok
test asset::model::tests::parse_rom_asset ... ok
test asset::texture::tests::parse_rom_asset ... ok
test asset::sound::tests::parse_rom_asset ... ok

test result: ok. 7 passed; 0 failed; 0 ignored; 0 measured; 10 filtered out; finished in 42.60s

     Running unittests src/main.rs (target/release/deps/cli-d252cf4ddfb6c3c0)

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

   Doc-tests ashen

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 1 filtered out; finished in 0.00s
```

**Comment by nenikitov** [comment]
I think there is some git weirdness going on, getting a fresh clone might fix it.

---

I noticed we have different hashes (or checksums, I didn't find anything on what it is) in the test outputs. Here are mine:
```
     Running unittests src/lib.rs (target/release/deps/ashen-f5441b6a101dccaf)
     Running unittests src/main.rs (target/release/deps/cli-319dd1f8ed4e0432)
```

I checked, these hashes `f544...` and `319d...` are the same on 2 different computers with different rust versions.

If I check out to the commit before the PR #57, I get your hashes
```sh
git checkout 83a5abc9672161992d33a9abe06240384835354c && cargo test --release -- --ignored parse_rom_asset
```
```
HEAD is now at 83a5abc chore: update to edition 2024 (#55)
    Finished `release` profile [optimized] target(s) in 3.33s
     Running unittests src/lib.rs (target/release/deps/ashen-bd671703daa93388)
     Running unittests src/main.rs (target/release/deps/cli-d252cf4ddfb6c3c0)
```

**Comment by RandomTale** [comment]
ooh yes that was the problem, after downloading from the main repository i got all the models
although im very sure i downloaded from there the previous time aswell, either that or it was from UserIsntAvalible. but even then i remember on top it said "its upto date with nenikotov's branch" or something , (alert) no i just checked it again to make sure and it says "1 commit behind", 🥲🥲 i feel very stupid wasting your time like this, perhaps i had made a mistake and didnt check it cuz im not used to using github, i literally had to check what stuff like fork and branch means, so yeah thanks a lot, ill use these in blender now hopefully to make some animation
